NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion
# Ah delete tag 15

# Description

As an admin I would like to be able to delete a Tag so that I can remove any that are not needed.
Given an admin is vising the list of Tags
When they select the Delete option
Then they should be prompted to confirm the deletion
Given the admin wishes to confirm the delete
When they the select the option to confirm
Then the Tag should be removed from the system
And the admin should be directed back to the Tag list
Given the admin decides not to confirm the delete
When they the select the option to reject confirmation
Then the Tag should NOT be removed from the system
And the admin should be directed back to the Tag list



**Created the following files and methods:**

- DeleteTag()
- DeleteTagById()
-”Delete” button


**Related Ticket(s)**
https://trello.com/c/s6axJtVP

#15 Delete a Tag

**Type of Change**


# Testing Instructions

#Setup (can be skipped if you already have the Tabloid database configured with provided seed data)
- Clone this project’s repository to your machine
  - If you’ve already done this and are currently working on a different branch of the project, be sure to `git add` and `git commit` whatever you’re working on on your branch, before you attempt to switch branches
  - Then `git checkout main` to switch to the project’s main branch
- Navigate to the root directory for this repo (yourParentFolderName/tabloidfullstack-the-body-slam-batallion)
- `git fetch -a` to retrieve the current remote-tracked branches for the project, you should see a branch named `AH-deleteTag-15` as one of the available branches
- `git checkout AH-deleteTag-15` to switch to my branch
- If you don't see the most recent version of my files, `git pull origin AH-deleteTag-15`
- `Start TabloidMVC.sln` to open the project in Visual Studio
- If you haven’t created the Tabloid database, please do so in Visual Studio and use the following SQL scripts to populate it accordingly:
  - Start a new query in your database and copy in this SQL script before running to fill out tables and columns: [First SQL Script for Creating DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/01_Tabloid_Create_DB.sql)
  - ctrl + a and delete what is in that query and replace it with this SQL script before running to seed the database: [Second SQL Script for Seeding DB](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-body-slam-batallion/blob/main/SQL/02_Tabloid_Seed_Data.sql)


- You will be presented with a login screen. Enter the email foo@bar.com and click "Login" (no password required)

- You will be presented with the welcome screen that simply says "hello"

- Click the Tag Management link in the topside header

-You should see a table that contains all Tags from your database as well as their corresponding Id. This table should be in alphabetical order by Tag Name

-To the right of  each item on the list you should see a “Delete” button

-Click said button on the tag named ”Funny”

-You will then be asked if you really want to delete the tag.

-select “Cancel.

- you will be redirected to the list of tags.

-Click “Delete” button on the tag named ”Funny” again.

-This time, select delete when prompted 

-You should see an updated list with “Funny” removed

-Run this SQL query:

```
SELECT * FROM Tag;
```
 to confirm that your “Funny” tag has been removed from the database.


# Checklist:
